### PR TITLE
[FIX] web: fix public env by introducing common env

### PR DESCRIPTION
**Pull request by seb-odoo**
A lot of keys were missing and services were not instantiated as expected.

### CODE REVIEW

**Comment by seb-odoo** [comment]
For our branch, should be backported in master/ensure it is done as they expected so we don't have to maintain an incorrect diff on those files.

File: addons/web/static/src/js/public/public_env.js [hunk 1]
```diff
@@ -1,24 +1,6 @@
...
-    // See https://github.com/odoo/owl/blob/master/doc/reference/environment.md#content-of-an-environment
-    // for more information on environments.
-    return {
-        qweb,
-        services: {
-            rpc: performRPC,
-        },
-    };
+    const env = require('web.common_env');
+    return env;
```
Comment (alexkuhn) [inline addons/web/static/src/js/public/public_env.js:6]: newline above

File: addons/web/static/src/js/public/public_root.js [hunk 2]
```diff
@@ -111,7 +112,10 @@ var PublicRoot = publicWidget.RootWidget.extend(ServiceProviderMixin, {
         } else if (ev.data.service === 'ajax' && ev.data.method === 'loadLibs') {
```
Comment (alexkuhn) [inline addons/web/static/src/js/public/public_root.js:113]: remove `@override` tag of function `_call_service`

File: addons/web/static/src/js/public/public_root.js [hunk 3]
```diff
@@ -25,12 +25,14 @@ var localeDef = ajax.loadJS('/web/webclient/locale/' + lang.replace('-', '_'));
...
  */
-var PublicRoot = publicWidget.RootWidget.extend(ServiceProviderMixin, {
+var PublicRoot = publicWidget.RootWidget.extend({
+    env: publicEnv,
     events: _.extend({}, publicWidget.RootWidget.prototype.events || {}, {
         'submit .js_website_submit_form': '_onWebsiteFormSubmit',
         'click .js_disable_on_click': '_onDisableOnClick',
     }),
     custom_events: _.extend({}, publicWidget.RootWidget.prototype.custom_events || {}, {
+        'call_service': '_call_service',
```
Comment (alexkuhn) [inline addons/web/static/src/js/public/public_root.js:35]: Component/widgets use this flow for service call, so they handle extension of global context with RPCs.
This doesn't seem the case with patched services in `env`... Not sure whether this is a problem or not?

(when entities make `rpc`, they rely on messaging service `rpc`, so in theory this affects almost all our RPCs in this refactoring)

File: addons/web/static/src/js/public/public_root.js [hunk 4]
```diff
@@ -111,7 +112,10 @@ var PublicRoot = publicWidget.RootWidget.extend(ServiceProviderMixin, {
         } else if (ev.data.service === 'ajax' && ev.data.method === 'loadLibs') {
             ev.data.args[1] = _computeContext.call(this, ev.data.args[1]);
         }
-        return ServiceProviderMixin._call_service.apply(this, arguments);
+        const service = this.env.services[ev.data.service];
+        const result = service[ev.data.method].apply(service, ev.data.args);
```
Comment (alexkuhn) [inline addons/web/static/src/js/public/public_root.js]: `service[ev.data.method](...ev.data.args)`

**Review by alexkuhn — COMMENTED** [review]
didn't made comment on code moved from `env` to `common_env`, because I think this is out of scope of this PR

**Comment by seb-odoo** [comment]
In progress by JS in master + livechat won't be converted yet

**Pull request closed without merging**